### DISCUSSION TITLE
Duda con el for en la kata 4

### DISCUSSION THREAD

**Discussion by JhosianGtz** [LaunchX-InnovaccionVirtual/CursoIntroPython#161] [Q&A] [answered]
Muy buenas, tengo una duda sobre el funcionamiento del ciclo for en la kata 4, no entiendo muy bien que hace.

**Comment by Paulina-Ceciliano** [comment, marked as answer]
Lo mejor que puedes hacer para entenderlo es ver la sesión del miércoles, Fer explica todo en ese live y lo explica super bien!
https://teams.microsoft.com/_#/broadcastPlaybackScreen

**Reply by JhosianGtz** [reply]
Muchas gracias eso haré

**Comment by MSuarez0** [comment]
la funcion FOR lo que hace es un bucle, y lo que hace mas o menos es depende de lo que quieras, hara el bucle y no saldras de hay hasta que termine la operacion, saldra de ese bucle y seguira el proceso.
ejemplo en esta imagen
![ciclo for (1)](https://user-images.githubusercontent.com/99224742/153693264-687a393b-afdd-40ff-8d4b-039199254bf2.png)

**Reply by JhosianGtz** [reply]
Si se como funciona el for y el while, pero no entendí bien lo que hace el de la kata 4 :c

**Reply by MSuarez0** [reply]
en este caso, este for, seguira el proceso hasta que  cuente hasta 15 y al llegar al quince llegara al final de ese bucle.

**Reply by OscarDelgad0** [reply]
¿En los ciclos anidados?

**Reply by JhosianGtz** [reply]
El numero 15 es por la cantidad de palabras en la frase?

**Reply by JhosianGtz** [reply]
> 

Sii en ese

**Comment by richirrim** [comment]
Lo que te recomiendo es usar mucha la función print() de python y siempre que no entiedas que esta haciendo algún fragmento de código, imprimas la salida y así iras entendiendo poco a poco. Te dejo un ejemplo.

lo que hace es devolverte cada elemento dentro de la lista.

  ```
animals = ['dog', 'cat', 'bow']

for animal in animals:
      print(animal)

# salida1: dog
# salida2: cat
# salida3: bow

**Comment by MSuarez0** [comment]
en este caso de la KATA 4, es que primero

for sentence in text_parts: ## setence es un nueva variable, entonces aqui tomara la primera oracion cortada anterior mente, cuando termine el primer ciclo, tomara la siguiente oracion
    for key_word in key_words: ## key_word tambien es una nueva variable que tomara la primera palabra que buscara, y cuando termine de hacer el primer ciclo del for, tomara la segunda palabra y la seguira buscando en la oracion.
        if key_word in sentence: ## aqui tomara la primera palabara a buscar y la buscara en la oracion, 
            print(sentence)
            break

y terminara los for cuando ya haya buscado todas las palabras en todas las oraciones que cortates

**Comment by alanmoren0** [comment]
Ese tipo de for es conocido como for each, si se tiene una lista puedes recorrer el número total de elementos dentro de esta lista a recorrer.
En este caso recorremos la totalidad de la lista "**text_parts**" y cada iteración de este va en la variable "**sentence**".
A la vez repetimos el proceso en la lista "**key_words**".
Es decir: en la primera iteración tenemos el elemento _sentence_ en la posición 0 que es equivalente _text_parts_ en la posición 0 y así de manera "automática" tendremos un incremento en +1 para cada elemento de la lista y esto se aplica de igual manera para el siguiente for anidado.

**Reply by alanmoren0** [reply]
Creo que no me di a entender muy bien.
Es cómo su estructura es:
for **sentence** in **text_parts**
Como lo entendería es.
**sentence** una variable temporal en la que va un elemento de la lista y que pasa al siguiente elemento en cada iteración
Y **text_parts** es la lista a recorrer

**Comment by JhosianGtz** [comment]
Ya me quedó más claro, muchas gracias a todos por tomarse el tiempo de responderme.
